tfhe/src/core_crypto/algorithms/seeded_lwe_ciphertext_list_decompression.rs

```diff
@@ -50,6 +52,61 @@ pub fn decompress_seeded_lwe_ciphertext_list_with_existing_generator<
     }
 }
 
+/// Convenience function to share the core logic of the decompression algorithm for
+/// [`SeededLweCiphertextList`] between all functions needing it.
+pub fn par_decompress_seeded_lwe_ciphertext_list_with_existing_generator<
+    Scalar,
+    InputCont,
+    OutputCont,
+    Gen,
+>(
+    output_list: &mut LweCiphertextList<OutputCont>,
+    input_seeded_list: &SeededLweCiphertextList<InputCont>,
+    generator: &mut RandomGenerator<Gen>,
+) where
+    Scalar: UnsignedTorus + Send + Sync,
+    InputCont: Container<Element = Scalar>,
+    OutputCont: ContainerMut<Element = Scalar>,
+    Gen: ParallelByteRandomGenerator,
+{
+    assert_eq!(
+        output_list.ciphertext_modulus(),
+        input_seeded_list.ciphertext_modulus(),
+        "Mismatched CiphertextModulus \
+    between input SeededLweCiphertextList ({:?}) and output LweCiphertextList ({:?})",
+        input_seeded_list.ciphertext_modulus(),
+        output_list.ciphertext_modulus(),
+    );
+
+    let ciphertext_modulus = output_list.ciphertext_modulus();
+
+    let bytes_per_child = (Scalar::BITS / 8) * (output_list.lwe_size().to_lwe_dimension().0);
+    let generators = generator
+        .par_try_fork(output_list.lwe_ciphertext_count().0, bytes_per_child)
+        .expect("Failed to fork generator");
```

Comment:
there should be a par_fork_lwe_list_to_lwe that does that already, we should probably update the non parallel version above as well

Reply:
I did found a `par_fork_lwe_list_to_lwe` but it seems to only be available on `EncryptionGenerator` not `RandomGenerator` that we have here

Reply:
Ah right, that case may not have been predicted initially

Reply:
One of the current silent invariant is that mask byte gen does not need additional bytes as we don’t do rejection sampling, so a generator used sequentially does the job, the current pattern is forking the generator for rayon processing, we may want to have an intermediate generator MaskGenerator and NoiseGenerator which have the forking primitives and then wrap those in the encryption random generator as is currently the case and forward forking calls to the underlying new Mask and Noise gens. I guess a todo/backlog issue will do for now :/

Reply:
Not sure I understood everything but ok 😅